generate/genfile/assert_test.go

```diff
@@ -0,0 +1,196 @@
+// Copyright 2022 Mineiros GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package genfile_test
+
+import (
+	"testing"
+
+	"github.com/mineiros-io/terramate/config"
+	"github.com/mineiros-io/terramate/errors"
+	"github.com/mineiros-io/terramate/hcl/eval"
+	. "github.com/mineiros-io/terramate/test/hclutils"
+	. "github.com/mineiros-io/terramate/test/hclwrite/hclutils"
+)
+
+func TestGenerateFileAssert(t *testing.T) {
+	t.Parallel()
+
+	tcases := []testcase{
+		{
+			name:  "multiple assert configs accessing metadata/globals/lets",
+			stack: "/stack",
+			configs: []hclconfig{
+				{
+					path: "/stack/globals.tm",
+					add: Globals(
+						Str("a", "value"),
+					),
+				},
+				{
+					path: "/stack/generate.tm",
+					add: GenerateFile(
+						Labels("asserts.hcl"),
+						Lets(
+							Expr("a", "global.a"),
+						),
+						Assert(
+							Expr("assertion", "let.a == global.a"),
+							Str("message", "let.a != global.a"),
+						),
+						Assert(
+							Expr("assertion", `terramate.stack.path.absolute == "/stack"`),
+							Str("message", "wrong stack metadata"),
+							Bool("warning", true),
+						),
+						Expr("content", "let.a"),
+					),
+				},
+			},
+			want: []result{
+				{
+					name: "asserts.hcl",
+					file: genFile{
+						origin:    "/stack/generate.tm",
+						condition: true,
+						body:      "value",
+						asserts: []config.Assert{
+							{
+								Range:     Mkrange("/stack/generate.tm", Start(7, 17, 88), End(7, 34, 105)),
```

Comment:
maybe put the filename in a `const` (or use a helper func) and reuse here to avoid repetition, but feel free to  ignore.